crates/trie/src/metrics.rs

```diff
@@ -0,0 +1,69 @@
+use metrics::Histogram;
+use reth_metrics::Metrics;
+use std::time::Instant;
+
+#[derive(Metrics)]
+#[metrics(scope = "trie")]
+pub(crate) struct TrieRootMetrics {
+    /// The number of microseconds trie root calculation lasted.
+    duration_micros: Histogram,
+    /// The number of branches added during trie root calculation.
+    branches_added: Histogram,
+    /// The number of leaves added during trie root calculation.
+    leaves_added: Histogram,
+}
+
+impl TrieRootMetrics {
+    pub(crate) fn new(ty: TrieType) -> Self {
+        Self::new_with_labels(&[("type", ty.as_str())])
```

Comment:
ideally, we should cache this initialization that, see https://github.com/paradigmxyz/reth/pull/6730.

Reply:
#6730 was impactful because metrics initialization was on a hot code path. not the case here, so i wouldn't worry too much